rocAL/rocAL/source/api/rocal_api_data_transfer.cpp

```diff
@@ -28,58 +28,20 @@ THE SOFTWARE.
 #endif
 
 RocalStatus ROCAL_API_CALL
-rocalToTensor32(RocalContext p_context, float *out_ptr, RocalTensorLayout tensor_format, float multiplier0,
+rocalToTensor(RocalContext p_context, void *out_ptr, RocalTensorLayout tensor_format, RocalTensorOutputType tensor_output_type, float multiplier0,
                        float multiplier1, float multiplier2, float offset0, float offset1, float offset2,
                        bool reverse_channels, RocalOutputMemType output_mem_type)
 {
     auto context = static_cast<Context*>(p_context);
     try
     {
-        auto tensor_layout = (tensor_format == ROCAL_NHWC) ?  RocalTensorFormat::NHWC : RocalTensorFormat::NCHW;
-        //auto tensor_output_data_type = (tensor_data_type == ROCAL_FP32) ? RocalTensorDataType::FP32 : RocalTensorDataType::FP16;
-        context->master_graph->to_tensor(out_ptr, tensor_layout, multiplier0, multiplier1, multiplier2,
-                offset0, offset1, offset2, reverse_channels, RocalTensorDataType::FP32, output_mem_type);
-    }
-    catch(const std::exception& e)
-    {
-        context->capture_error(e.what());
-        ERR(e.what())
-        return ROCAL_RUNTIME_ERROR;
-    }
-    return ROCAL_OK;
-}
+        if(tensor_format != ROCAL_NHWC && tensor_format != ROCAL_NCHW)
+            THROW("Supported only for NHWC and NCHW tensor layout")
```

Comment:
why the implementation is removed here?

Reply:
There are 3 API, rocalToTensor16, rocalToTensor32 and rocalToTensor which internally calls the same to_tensor function internally and the only difference is the output type.
We have removed rocalToTensor16, rocalToTensor32 and retained only rocalToTensor so we can have a unified API for this case